sdk/keyvault/keyvault/test/client_key_poc.c

```diff
@@ -58,6 +58,7 @@ int main() {
   // init credential_credentials struc
   az_result creds_retcode = az_client_secret_credential_init(
       &credential,
+      AZ_STR("https://vault.azure.net"),
```

Comment:
this I don't really like.
The code in this file `client_key_poc.c` is what a customer would actually write. I don't like customer having to know about what resource to use to create a credential

Reply:
I decoupled resource from the credential, and going to make service clients to provide the resource/scopes.